packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,350 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./BridgePoolFactoryInterface.sol";
+import "../../../contracts/oracle/interfaces/OptimisticOracleInterface.sol";
+import "../../../contracts/oracle/interfaces/StoreInterface.sol";
+import "../../../contracts/oracle/interfaces/FinderInterface.sol";
+import "../../../contracts/oracle/implementation/Constants.sol";
+import "../../../contracts/common/implementation/AncillaryData.sol";
+import "../../../contracts/common/implementation/Testable.sol";
+import "../../../contracts/common/implementation/FixedPoint.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 /**
- * @notice Contract deployed on L1 that ingests liquidity from passive liquidity providers and returns them claims to
- * withdraw their funds.
+ * @notice Contract deployed on L1 that provides methods for "Relayers" to fulfill deposit orders that originated on L2.
+ * The Relayers can either post capital to fulfill the deposit instantly, or request that the funds are taken out of
+ * a passive liquidity provider pool following a challenge period. Related, this contract ingests liquidity from
+ * passive liquidity providers and returns them claims to withdraw their funds. Liquidity providers are incentivized
+ * to post collateral by earning a fee per fulfilled deposit order.
+ * @dev A "Deposit" is an order to send capital from L2 to L1, and a "Relay" is a fulfillment attempt of that order.
  */
-abstract contract BridgePool is ERC20 {
+contract BridgePool is Testable {
+    using SafeERC20 for IERC20;
+    using FixedPoint for FixedPoint.Unsigned;
+
+    // Administrative contract that deployed this contract and also houses all state variables needed to relay deposits.
+    BridgePoolFactoryInterface public bridgePoolFactory;
+
+    // A Deposit represents a transfer that originated on an L2 DepositBox contract and can be bridged via this contract.
+    enum DepositState { Uninitialized, PendingSlow, PendingInstant, FinalizedSlow, FinalizedInstant }
+    enum DepositType { Slow, Instant }
+
+    // @dev: There is a limit to how many params a struct can contain. Without encapsulating some of the Deposit params
+    // inside the RelayAncillaryDataContents struct, the compiler throws an error related to this issue:
+    // https://github.com/ethereum/solidity/issues/10930.
+    struct RelayAncillaryDataContents {
+        uint256 depositId;
+        // The following params are inferred by the L2 deposit:
+        address l2Sender;
```

Comment:
should this not be `l2Depositor`?

Reply:
Just a style thing. Before I make sweeping changes do you think it should be `l2Depositor`? Or even `depositor`?

Reply:
I guess `depositor` is the cleanest and most descriptive. However, looking on the `DepositBox` we still call it a `sender` so perhaps leave it for now and then we can make a sweeping PR for this kind of thing in a follow-up to make sure everything is consistent over all the contracts and actors in the system.